RecoTracker/PixelSeeding/plugins/alpaka/CAPixelDoublets.h

```diff
@@ -13,60 +13,47 @@
-          cellNeighbors->construct(TrackerTraits::maxNumOfActiveDoublets, cellNeighborsContainer);
-          cellTracks->construct(TrackerTraits::maxNumOfActiveDoublets, cellTracksContainer);
-          [[maybe_unused]] auto i = cellNeighbors->extend(acc);
-          ALPAKA_ASSERT_ACC(0 == i);
-          (*cellNeighbors)[0].reset();
-          i = cellTracks->extend(acc);
-          ALPAKA_ASSERT_ACC(0 == i);
-          (*cellTracks)[0].reset();
-        }
+                                    uint32_t maxNumOfDoublets,
+                                    CASimpleCell<TrackerTraits>* cells,
+                                    uint32_t* nCells,
+                                    HitsConstView hh,
+                                    ::reco::CAGraphSoAConstView cc,
+                                    ::reco::CALayersSoAConstView ll,
+                                    uint32_t const* __restrict__ offsets,
+                                    PhiBinner<TrackerTraits> const* phiBinner,
+                                    HitToCell* outerHitHisto,
+                                    AlgoParams const& params) const {
+        doubletsFromHisto<TrackerTraits>(
+            acc, maxNumOfDoublets, cells, nCells, hh, cc, ll, offsets, phiBinner, outerHitHisto, params);
       }
     };
 
-    // Not used for the moment, see below.
-    //constexpr auto getDoubletsFromHistoMaxBlockSize = 64;  // for both x and y
-    //constexpr auto getDoubletsFromHistoMinBlocksPerMP = 16;
-
     template <typename TrackerTraits>
-    class GetDoubletsFromHisto {
+    class FillDoubletsHisto {
     public:
       template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
-      // #ifdef __CUDACC__
-      //       __launch_bounds__(getDoubletsFromHistoMaxBlockSize, getDoubletsFromHistoMinBlocksPerMP)  // TODO: Alapakify
-      // #endif
       ALPAKA_FN_ACC void operator()(TAcc const& acc,
-                                    CACellT<TrackerTraits>* cells,
+                                    CASimpleCell<TrackerTraits> const* __restrict__ cells,
                                     uint32_t* nCells,
-                                    CellNeighborsVector<TrackerTraits>* cellNeighbors,
-                                    CellTracksVector<TrackerTraits>* cellTracks,
-                                    HitsConstView<TrackerTraits> hh,
-                                    OuterHitOfCell<TrackerTraits>* isOuterHitOfCell,
-                                    uint32_t nActualPairs,
-                                    const uint32_t maxNumOfDoublets,
-                                    CellCutsT<TrackerTraits> cuts) const {
-        doubletsFromHisto<TrackerTraits>(
-            acc, nActualPairs, maxNumOfDoublets, cells, nCells, cellNeighbors, cellTracks, hh, *isOuterHitOfCell, cuts);
+                                    uint32_t offsetBPIX2,
+                                    HitToCell* outerHitHisto) const {
```

Comment:
random comment, can this be `__restrict__` as well, or does it potentially alias the other parameters ?